### PR TITLE
Map metric names based on description

### PR DESCRIPTION
**Pull request by sdmarshall79**

### CODE REVIEW

File: internal/metrics/update.go [hunk 1]
```diff
@@ -168,13 +182,15 @@ func updateMetrics(metrics map[string]*metricData) {
...
+					metric, ok := metrics[makeKey(metricElement)]
+					if ok {
+						metric.values = make(map[string]float64)
+
+						// Update metric with cached values of publication data
+						for label, value := range metricElement.Values {
+							normalisedValue := mqmetric.Normalise(metricElement, label, value)
+							metric.values[label] = normalisedValue
+						}
 					}
```
Comment (parrobe) [inline internal/metrics/update.go]: What about if we're not ok? Do we need an error. What does it mean?
Can we either have an else with error handling or comment as to why it doesn't matter.

File: internal/metrics/update.go [hunk 2]
```diff
@@ -187,5 +203,115 @@ func updateMetrics(metrics map[string]*metricData) {
 
 // makeKey builds a unique key for each metric
 func makeKey(metricElement *mqmetric.MonElement) string {
-	return metricElement.Parent.Parent.Name + "/" + metricElement.Parent.Name + "/" + metricElement.MetricName
+	return metricElement.Parent.Parent.Name + "/" + metricElement.Parent.Name + "/" + metricElement.Description
+}
+
+// generateMetricNamesMap generates metric names mapped from their description
+func generateMetricNamesMap() map[string]string {
```
Comment (parrobe) [inline internal/metrics/update.go]: Might be worth moving this function into a separate file?

File: internal/metrics/update.go [hunk 3]
```diff
@@ -187,5 +203,115 @@ func updateMetrics(metrics map[string]*metricData) {
...
+		{"STATMQI/GET/Persistent message destructive get - count", "persistent_message_destructive_get_count"},
+		{"STATMQI/GET/Non-persistent message browse - byte count", "non_persistent_message_browse_bytes"},
+		{"STATMQI/GET/Persistent message browse - byte count", "persistent_message_browse_bytes"},
+		{"STATMQI/GET/MQCB count", "mqcb_count"},
+		{"STATMQI/GET/Failed MQCB count", "failed_mqcb_count"},
+		{"STATMQI/SYNCPOINT/Commit count", "commit_count"},
+		{"STATMQI/SYNCPOINT/Rollback count", "rollback_count"},
+	}
+
+	for _, mapping := range mappings {
```
Comment (parrobe) [inline internal/metrics/update.go]: You don't have to do this. you can initialize golang maps with data.
 To initialize a map with some data, use a map literal:
```
commits := map[string]int{
    "rsc": 3711,
    "r":   2138,
    "gri": 1908,
    "adg": 912,
}
```

File: test/docker/mqmetric_test.go [hunk 4]
```diff
@@ -387,7 +387,7 @@ func TestChangingValues(t *testing.T) {
 	conn.Close()
 
 	// Now actually get the metrics (after waiting for some to become available)
-	time.Sleep(15 * time.Second)
+	time.Sleep(25 * time.Second)
 	metrics = getMetrics(t, port)
 	if len(metrics) <= 0 {
 		t.Fatal("Expected some metrics to be returned but had none...")
```
Comment (parrobe) [inline test/docker/mqmetric_test.go]: Do we need a new test that checks all the metrics returned are known about and none have say changed name or we have had 1 removed and a new one added so the total remains the same?
